File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float get_positive_float(void);

int main(void)
{
    int cents = round(get_positive_float() * 100);
    int i = 0;
    while (cents > 0)
    {
        if ((cents - 25) >= 0)
        {
            cents = cents - 25;
        }
        else if ((cents - 10) >= 0)
        {
            cents = cents - 10;
        }
        else if ((cents - 5) >= 0)
        {
            cents = cents - 5;
        }
        else
        {
            cents = cents - 1;
        }
        i++;
    }
    printf("%d\n", i);
}


float get_positive_float(void)
{
    float n;
    do
    {
        n = get_float("%s", "Change owed: ");
    }
    while (n <= 0);
    return n;
}

File: pset6/cash/cash.py
from cs50 import get_float
from math import floor


def main():
    changeOwed = get_float("Change owed: ")
    if changeOwed < 0:
        return main()
    quarters = floor(changeOwed / 0.25)
    dimes = floor(round(changeOwed % 0.25, 2) / 0.1)
    nickels = floor(round(round(changeOwed % 0.25, 2) % 0.1, 2) / 0.05)
    pennies = floor(round(round(round(changeOwed % 0.25, 2) %
                                0.1, 2) % 0.05, 2) / 0.01)
    print(quarters + dimes + nickels + pennies)


if __name__ == "__main__":
    main()